### PR TITLE
Feature: Pet Nametag

### PR DESCRIPTION
**Pull request by j10a1n15**
## Dependencies
- #1687 

## What
Adds options to modify the look of pet nametags, like ~~removing the player name, or~~ hiding the level of the pet.

<details>
<summary>Images</summary>

image

</details>

## Changelog New Features
+ Added option to hide pet nametag text. - Empa + j10a1n15

### CODE REVIEW

**Comment by lunaynx** [comment]
You aren't going to guess what's on alpha right now...
image

File: src/main/java/at/hannibal2/skyhanni/features/misc/pets/PetNametag.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+object PetNametag {
+
+    private val config get() = SkyHanniMod.feature.misc.pets.petNametag
+
+    /**
+     * REGEX-TEST: §8[§7Lv99§8] Ammonite
+     * REGEX-TEST: §8[§7Lv100§8] Endermite§5 ✦
+     */
+    private val pattern by RepoPattern.pattern(
+        "feature.pet.nametag",
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/misc/pets/PetNametag.kt]: feature feels weird as the start of the pattern key

File: src/main/java/at/hannibal2/skyhanni/features/misc/pets/PetNametag.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        "(?<start>§8\\[§7Lv(?<lvl>\\d+)§8\\]) (?<rarity>§.)(?<pet>[\\w\\s]+)(?<skin>§. ✦)?"
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/misc/pets/PetNametag.kt]: not super important but 
Redundant character escape '\\]' in RegExp

File: src/main/java/at/hannibal2/skyhanni/features/misc/pets/PetNametag.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+@SkyHanniModule
+object PetNametag {
+
+    private val config get() = SkyHanniMod.feature.misc.pets.petNametag
+
+    /**
+     * REGEX-TEST: §8[§7Lv99§8] Ammonite
+     * REGEX-TEST: §8[§7Lv100§8] Endermite§5 ✦
+     */
+    private val pattern by RepoPattern.pattern(
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/misc/pets/PetNametag.kt]: while there is only one pattern is probably better to name the variable something more than just pattern

File: src/main/java/at/hannibal2/skyhanni/features/misc/pets/PetNametag.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            if (!config.hidePetLevel && (!config.hideMaxPetLevel || lvl == 100 || lvl == 200)) text += "$start "
+            text += rarity + pet + skin
+
+            event.chatComponent = text.asComponent()
+        }
+    }
+
+    private fun isEnabled() = with(config) {
+        (hidePetLevel || hideMaxPetLevel) && LorenzUtils.inSkyBlock
+    }
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/misc/pets/PetNametag.kt]: should check if in skyblock first, also im not sure if we need to be doing with(config) as we could do it the way its done everywhere else, but thats up to you

File: src/main/java/at/hannibal2/skyhanni/data/EntityData.kt [hunk 5]
```diff
@@ -98,7 +98,7 @@ object EntityData {
         ci.returnValue = postRenderNametag(entity, ci.returnValue)
     }
 
-    private fun postRenderNametag(entity: Entity, chatComponent: IChatComponent) = nametagCache.getOrPut(entity) {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/data/EntityData.kt]: not sure why this changed

File: src/main/java/at/hannibal2/skyhanni/mixins/transformers/MixinEntity.java [hunk 6]
```diff
@@ -13,6 +13,6 @@ public abstract class MixinEntity {
 
     @Inject(method = "getDisplayName", at = @At("RETURN"), cancellable = true)
     public void getDisplayName(CallbackInfoReturnable<IChatComponent> ci) {
-        EntityData.getDisplayName((Entity) (Object) this, ci);
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/mixins/transformers/MixinEntity.java]: not sure why this changed x2

File: src/main/java/at/hannibal2/skyhanni/mixins/transformers/MixinEntityPlayer.java [hunk 7]
```diff
@@ -13,6 +13,6 @@ public class MixinEntityPlayer {
 
     @Inject(method = "getDisplayName", at = @At("RETURN"), cancellable = true)
     public void getDisplayName(CallbackInfoReturnable<IChatComponent> ci) {
-        EntityData.getDisplayName((EntityPlayer) (Object) this, ci);
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/mixins/transformers/MixinEntityPlayer.java]: not sure why this changed x3

File: src/main/java/at/hannibal2/skyhanni/config/features/misc/pets/PetNametagConfig.java
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/misc/pets/PetNametagConfig.java:1]: both of these should be next to the pet instead of above

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
Hiding the level worked for me, the hiding only maxed levels did not work, it only showed max leveled pets levels and hid all the other ones.

File: src/main/java/at/hannibal2/skyhanni/data/EntityData.kt (hunk 5 above)
Comment (j10a1n15) [reply src/main/java/at/hannibal2/skyhanni/data/EntityData.kt]: The old system only allowed to add onto the Entity Nametag, while this new one allows a full replacement.
Not sure why Empa didn't just change it in their original pr

Comment (ThatGravyBoat) [reply src/main/java/at/hannibal2/skyhanni/data/EntityData.kt]: it already allowed full replacement its just all the logic was in the hook, ie. EntityData you seemed to have moved this logic into the mixin, please revert the changes to this class and both mixins as it does the exact same thing.

**Review by ThatGravyBoat — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/config/features/misc/pets/PetNametagConfig.java
Comment (j10a1n15) [reply src/main/java/at/hannibal2/skyhanni/config/features/misc/pets/PetNametagConfig.java:1]: I have no idea what this means

**Review by CalMWolfs — APPROVED** [review]

**Pull request merged**